### PR TITLE
[NetAppFiles] Apply 2021-10-01 Changes

### PR DESCRIPTION
**Pull request by audunn**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 [assembly: AssemblyDescription("Provides management functionality for Azure NetApp Storage.")]
 
 [assembly: AssemblyVersion("1.0.0.0")]
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Properties/AssemblyInfo.cs:11]: Need to update `AssemblyVersion` to be consistent with the version number (e.g. `1.16.0.0`). Actually `AssemblyVersion` is used by runtime. See: https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/general/general/assembly-version-assembly-file-version

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs [hunk 2]
```diff
@@ -192,7 +196,7 @@ public partial interface IVolumesOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string accountName, string poolName, string volumeName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string accountName, string poolName, string volumeName, bool? forceDelete = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs:199]: There are breaking changes in the interfaces. Here are 2 options:
- Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
- Or you can consider to bump the major version to `2.0.0`

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/CapacityPoolPatch.cs [hunk 3]
```diff
@@ -108,25 +108,5 @@ public CapacityPoolPatch(string location = default(string), string id = default(
         [JsonProperty(PropertyName = "properties.qosType")]
         public string QosType { get; set; }
 
-        /// <summary>
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/CapacityPoolPatch.cs:111]: Breaking change

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/Volume.cs [hunk 4]
```diff
@@ -122,7 +125,12 @@ public Volume()
...
...aProtection = default(VolumePropertiesDataProtection), bool? isRestoring = default(bool?), bool? snapshotDirectoryVisible = default(bool?), bool? kerberosEnabled = default(bool?), string securityStyle = default(string), bool? smbEncryption = default(bool?), bool? smbContinuouslyAvailable = default(bool?), double? throughputMibps = default(double?), string encryptionKeySource = default(string), bool? ldapEnabled = default(bool?), bool? coolAccess = default(bool?), int? coolnessPeriod = default(int?), string unixPermissions = default(string), int? cloneProgress = default(int?), string avsDataStore = default(string), bool? isDefaultQuotaEnabled = default(bool?), long? defaultUserQuotaInKiBs = default(long?), long? defaultGroupQuotaInKiBs = default(long?), long? maximumNumberOfFiles = default(long?), string volumeGroupName = default(string), string capacityPoolResourceId = default(string), string proximityPlacementGroup = default(string), string t2Network = default(string), string volumeSpecName = default(string), IList<PlacementKeyValuePairs> placementRules = default(IList<PlacementKeyValuePairs>), string enableSubvolumes = default(string), SystemData systemData = default(SystemData))
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/Volume.cs:133]: breaking change in interface

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/VolumeGroupVolumeProperties.cs [hunk 5]
```diff
@@ -121,7 +124,10 @@ public VolumeGroupVolumeProperties()
...
...t(string), VolumePropertiesDataProtection dataProtection = default(VolumePropertiesDataProtection), bool? isRestoring = default(bool?), bool? snapshotDirectoryVisible = default(bool?), bool? kerberosEnabled = default(bool?), string securityStyle = default(string), bool? smbEncryption = default(bool?), bool? smbContinuouslyAvailable = default(bool?), double? throughputMibps = default(double?), string encryptionKeySource = default(string), bool? ldapEnabled = default(bool?), bool? coolAccess = default(bool?), int? coolnessPeriod = default(int?), string unixPermissions = default(string), int? cloneProgress = default(int?), string avsDataStore = default(string), bool? isDefaultQuotaEnabled = default(bool?), long? defaultUserQuotaInKiBs = default(long?), long? defaultGroupQuotaInKiBs = default(long?), long? maximumNumberOfFiles = default(long?), string volumeGroupName = default(string), string capacityPoolResourceId = default(string), string proximityPlacementGroup = default(string), string t2Network = default(string), string volumeSpecName = default(string), IList<PlacementKeyValuePairs> placementRules = default(IList<PlacementKeyValuePairs>), string enableSubvolumes = default(string))
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/VolumeGroupVolumeProperties.cs:130]: breaking change

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/VolumesOperations.cs [hunk 6]
```diff
@@ -650,16 +650,20 @@ public async Task<AzureOperationResponse<Volume>> UpdateWithHttpMessagesAsync(Vo
...
+        /// the particular volume
+        /// </param>
         /// <param name='customHeaders'>
         /// The headers that will be added to request.
         /// </param>
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        public async Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string accountName, string poolName, string volumeName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse> DeleteWithHttpMessagesAsync(string resourceGroupName, string accountName, string poolName, string volumeName, bool? forceDelete = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (archerzz) [inline sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/VolumesOperations.cs:663]: breaking change

**Review by archerzz — CHANGES_REQUESTED** [review]
A few breaking changes are detected. Basically there are two options:
- add customization codes
- bump to next major version

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Properties/AssemblyInfo.cs]: version bumped, we have been updating `AssemblyFileVersion` only will update `AssemblyVersion` from now on, thanks

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs (hunk 2 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs:199]: We bumped the major version, thanks

Comment (heaths) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs:199]: NuGet doesn't really support semver like that. For track 2, we have a strict no-breaking changes policy and bumping the major version won't solve breaking changes. Instead, you can add an overloads. See https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-parameters for several tips/requirements how to do that. You can also hide the older one using `[EditorBrowsable(EditorBrowsableState.Never)]`.

Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs:199]: Thanks for the points, we will make the suggested changes

**Review by archerzz — APPROVED** [review]

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/VolumeGroupVolumeProperties.cs (hunk 5 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/VolumeGroupVolumeProperties.cs:130]: VolumeGroup operations are hardly cause of breaking changes as it is a new feature/operations introduced in this version.

**Comment by heaths** [comment]
FWIW, the Azure SDK team started what is known as track 2 with several goals in mind based on user feedback. Some of the biggest feedback was that developers were tired of breaking changes - both in the SDK as well as the REST APIs. This is why we have a strict no-breaking changes policy. Adding overloads is an easy change to avoid this and I highly recommend doing that even for track 1. If the packages are in active use - and presumably they are since you're updating them - that they are track 1 doesn't change the feedback.

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/VolumesOperations.cs (hunk 6 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/VolumesOperations.cs:663]: customization added

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/Volume.cs (hunk 4 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/Models/Volume.cs:133]: customization added

File: sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs (hunk 2 above)
Comment (audunn) [reply sdk/netapp/Microsoft.Azure.Management.NetApp/src/Generated/IVolumesOperations.cs:199]: customization added

**Comment by heaths** [comment]
Right - if the SDK doesn't contain any breaking changes, I'd only bump the minor version.

**Comment by audunn** [comment]
@archerzz Customizations added for backwards compatibility.  Are we good to proceed?

**Comment by audunn** [comment]
@archerzz  Are we good to proceed?

**Pull request merged**